kong/tracing/propagation.lua

```diff
@@ -520,6 +520,42 @@ local function find_header_type(headers)
 end
 
 
+-- Adds trace ID formats to the current request's and returns a table
+-- containing all the formats.
+--
+-- Plugins can handle different formats of trace ids depending on their headers
+-- configuration, multiple plugins executions may result in additional formats
+-- of the current request's trace id.
+--
+-- Each item in the resulting `trace_id_all_fmt` table represents a
+-- format associated with the trace ID for the current request.
+--
+-- @param trace_id_new_fmt table containing the trace ID formats to be added
+-- @returns trace_id_all_fmt table contains all the formats for this request
+--
+-- @example
+--
+--    existing_formats = { datadog = "1234",
+--                         w3c     = "abcd" }
+--
+--    trace_id_new_fmt = { ot = "abcd", w3c = "abcd" }
+--
+--    trace_id_all_fmt = { datadog = "1234", ot = "abcd", w3c = "abcd" }
+--
+local function add_trace_id_formats(trace_id_new_fmt)
+  -- TODO: @samugi - move this in the unified tracing context
+  local trace_id_all_fmt = ngx.ctx.propagation_trace_id_all_fmt or {}
```

Comment:
can we improve the performance here, like adding this code segment at the start of this function? Then we dont copy all the items into the new created ngx.ctx.propagation_trace_id_all_fmt empty table. 

```
if not ngx.ctx.propagation_trace_id_all_fmt then
   ngx.ctx.propagation_trace_id_all_fmt = trace_id_new_fmt
   return
end
```